ca/ca.go

```diff
@@ -0,0 +1,51 @@
+package ca
+
+import (
+	"github.com/rightscale/rsc/ca/cac"
+	"github.com/rightscale/rsc/cmd"
+	"github.com/rightscale/rsc/metadata"
+	"github.com/rightscale/rsc/rsapi"
+	"regexp"
+)
+
+// Metadata synthetized from all CA APIs metadata
+var GenMetadata map[string]*metadata.Resource
+
+// CA 1.0 common client to all self-service APIs
+type Api struct {
+	*rsapi.Api
+}
+
+// Build client from command line
+func FromCommandLine(cmdLine *cmd.CommandLine) (*Api, error) {
+	api, err := rsapi.FromCommandLine(cmdLine)
+	if err != nil {
+		return nil, err
+	}
+	setupMetadata()
+	api.Host = apiHostFromLogin(cmdLine.Host)
+	api.Metadata = GenMetadata
+	return &Api{api}, nil
+}
+
+func apiHostFromLogin(host string) string {
+	integration, _ := regexp.MatchString("^cobalt", host)
+	staging, _ := regexp.MatchString("^moo", host)
+
+	switch {
```

Comment:
Is it possible to write generic code like the one for the self-service host instead? (capture the relevant prefix and suffix using a regexp)